schemas/com.roku/device_info/jsonschema/1-0-0

```diff
@@ -0,0 +1,159 @@
+      ],
+      "description": "A unique device identifier that is different across channels.",
+      "maxLength": 255
+    },
+    "isRIDADisabled": {
+      "type": [
+        "boolean",
+        "null"
+      ],
+      "description": "Indicates whether tracking via Roku's ID for Advertisers (RIDA) is disabled on the device."
+    },
+    "RIDA": {
+      "type": [
+        "string",
+        "null"
+      ],
+      "description": "A persistent unique identifier (UUID) for the device.",
+      "maxLength": 255
+    },
+    "captionsMode": {
+      "type": "string",
+      "description": "Determines whether global captions are turned on or off, or are in instant replay mode.",
+      "maxLength": 255
+    },
+    "audioOutputChannel": {
+      "type": "string",
+      "description": "Type of audio output.",
+      "maxLength": 255
+    },
+    "memoryLevel": {
+      "type": [
+        "string",
+        "null"
+      ],
+      "description": "General memory level of the channel.",
+      "maxLength": 255
+    },
+    "timeSinceLastKeypress": {
+      "type": "integer",
+      "description": "The number of seconds since the last remote keypress was received.",
+      "minimum": 0,
+      "maximum": 2147483647
+    },
+    "userCountryCode": {
+      "type": [
+        "string",
+        "null"
+      ],
+      "description": "ISO 3166-1 (2-letter) country code associated with the user's Roku account.",
+      "maxLength": 255
+    },
+    "countryCode": {
+      "type": "string",
+      "description": "Roku Channel Store associated with a user's Roku account.",
+      "maxLength": 255
+    },
+    "videoMode": {
+      "type": "string",
+      "description": "Video playback resolution.",
+      "maxLength": 255
+    },
+    "displayWidth": {
+      "type": "integer",
+      "description": "Physical width of the attached display in centimeters.",
+      "minimum": 0,
+      "maximum": 65535
+    },
+    "displayHeight": {
+      "type": "integer",
+      "description": "Physical height of the attached display in centimeters.",
+      "minimum": 0,
+      "maximum": 65535
+    },
+    "displayProperties": {
+      "type": "array",
+      "description": "List of keys for display properties of the screen.",
+      "items": {
+        "type": "string",
+        "description": "Display property of the screen.",
+        "maxLength": 255
+      }
+    },
+    "connectionType": {
+      "type": "string",
+      "description": "Type of internet connection the device is using.",
+      "maxLength": 255
+    },
+    "internetStatus": {
+      "type": [
+        "boolean",
+        "null"
+      ],
+      "description": "Internet connection status of the device."
+    },
+    "features": {
+      "type": "array",
+      "description": "List of features that the current device/firmware supports.",
+      "items": {
+        "type": "string",
+        "description": "Feature that the current device/firmware supports.",
+        "maxLength": 255
+      }
```

Comment:
Is this useful information? Is it a huge list? Pondering if its worth tracking. Could also be a flag in the tracker if it a big ol' list that not everyone will want.

Reply:
For my Roku, it only contains 1 item. It can get up to 10 items if the device supports everything. In the Roku SDK, these are implemented as individual booleans, but I thought putting them all into a list would be more compact and easier to extend in the future. I actually don't know if they are worth tracking, it's difficult for me to assess, but these are the features that it can include: 5.1_surround_sound, can_output_5.1_surround_sound, sd_only_hardware, usb_hardware, sdcard_hardware, ethernet_hardware, gaming_hardware, energy_star_compliant, soundbar_hardware, handsfree_voice

I would be okay with it if we decide not to include them at this time and wait for feedback if they are useful to anyone. The same goes for the displayProperties list.

Reply:
Lets included them if there's only a selection of them. Likely interesting metadata for people building video applications for Roku about what features are available.